### PR TITLE
DDCNL-513 | validation for nameOrNumber and postalCode

### PR DESCRIPTION
**Pull request by shikkhasingh**

### CODE REVIEW

File: test/unit/models/RequestFundReleaseRequestSpec.scala [hunk 1]
```diff
@@ -153,4 +155,72 @@ class RequestFundReleaseRequestSpec extends PlaySpec {
...
+    "nameOrNumber is Empty" in {
+
+      val invalidNameOrNumberJson = createJson("", "AA11 1AA")
+
+      extractErrorValidation(invalidNameOrNumberJson.validate[FundReleasePropertyDetails]) must contain(ErrorValidation("INVALID_NAME_OR_NUMBER", "Enter nameOrNumber", Some("/nameOrNumber")))
+    }
+
+    "nameOrNumber is too long" in {
+
+      val invalidNameOrNumberJson = createJson("aaaaaaaaaabbbbbbbbbbccccccccccdddddd", "AA11 1AA")
```
Comment (liampan) [inline test/unit/models/RequestFundReleaseRequestSpec.scala]: Please add a test for 35 chars exactly; I dont think your boundaries are what you expect them to be. :)

Comment (liampan) [inline test/unit/models/RequestFundReleaseRequestSpec.scala]: This is much easier to discern how many chars there are :)
```suggestion
      val invalidNameOrNumberJson = createJson("a" * 36 , "AA11 1AA")
```

Comment (shikkhasingh) [reply test/unit/models/RequestFundReleaseRequestSpec.scala]: Agreed
Changes done

Comment (shikkhasingh) [reply test/unit/models/RequestFundReleaseRequestSpec.scala]: Yeah this makes more sense
changes done

File: app/uk/gov/hmrc/lisaapi/models/RequestFundReleaseRequest.scala [hunk 2]
```diff
@@ -23,7 +23,48 @@ import play.api.libs.json._
...
+      }
+    }
+  }
+
+  private def postalCodeValidator: Reads[String] = new Reads[String] {
+    override def reads(postalCode: JsValue): JsResult[String] = {
+      postalCode.as[String] match {
+        case postalCode if postalCode.matches(postalCodeRegex) => JsSuccess(postalCode)
+        case postalCode if postalCode.isEmpty => toJsError("emptyPostalCode")
+        case postalCode => toJsError("invalidPostalCode")
```
[the remark this answers is not in the document]
Comment (shikkhasingh) [reply app/uk/gov/hmrc/lisaapi/models/RequestFundReleaseRequest.scala]: changes done

File: app/uk/gov/hmrc/lisaapi/models/RequestFundReleaseRequest.scala [hunk 3]
```diff
@@ -23,7 +23,48 @@ import play.api.libs.json._
...
+    nameOrNumber match {
+      case name if name.isEmpty => toJsError("emptyNameOrNumber")
+      case name if name.length > 35 => toJsError("tooLongNameOrNumber")
+      case _ => toJsError("invalidNameOrNumber")
+    }
+  }
+
+  private def nameOrNumberValidator: Reads[String] = new Reads[String] {
+    override def reads(nameOrNumber: JsValue): JsResult[String] = {
+      nameOrNumber.as[String] match {
```
[the remark this answers is not in the document]
Comment (shikkhasingh) [reply app/uk/gov/hmrc/lisaapi/models/RequestFundReleaseRequest.scala]: changes done

File: app/uk/gov/hmrc/lisaapi/models/RequestFundReleaseRequest.scala [hunk 4]
```diff
@@ -23,7 +23,48 @@ import play.api.libs.json._
...
+
+  private val postalCodeRegex = "^([A-Za-z][A-Ha-hK-Yk-y]?[0-9][A-Za-z0-9]? ?[0-9][A-Za-z]{2}|[Gg][Ii][Rr] ?0[Aa]{2})$"
+
+  private def toJsError(errorType : String) : JsError ={
+    JsError(JsonValidationError(errorType))
+  }
+
+  private def getNameOrNumberErrorMessage(nameOrNumber : String) : JsError = {
+    nameOrNumber match {
+      case name if name.isEmpty => toJsError("emptyNameOrNumber")
```
[the remark this answers is not in the document]
Comment (shikkhasingh) [reply app/uk/gov/hmrc/lisaapi/models/RequestFundReleaseRequest.scala]: Agreed but I think "case name if name.isEmpty" is more readable / understandable  than ""  .
